src/query/expression/src/type_check.rs

```diff
@@ -283,7 +283,7 @@ impl Subsitution {
         for (idx, ty2) in other.0 {
             if let Some(ty1) = self.0.remove(&idx) {
                 let common_ty = common_super_type(ty2.clone(), ty1.clone()).ok_or_else(|| {
-                    ErrorCode::SemanticError(format!(
```

Comment:
Backtrace is controlled by env.

```
pub fn capture() -> Backtrace {
      if !Backtrace::enabled() {
          return Backtrace { inner: Inner::Disabled };
      }
      Backtrace::create(Backtrace::capture as usize)
  }
```

Reply:
But we default to open this in ci tests.

```
# Enable backtrace to debug https://github.com/datafuselabs/databend/issues/7986
export RUST_BACKTRACE=full
./scripts/ci/deploy/databend-query-cluster-3-nodes.sh

SCRIPT_PATH="$(cd "$(dirname "$0")" >/dev/null 2>&1 && pwd)"
cd "$SCRIPT_PATH/../../tests" || exit

echo "Starting databend-test"
# 13_0004_q4: https://github.com/datafuselabs/databend/issues/8107
./databend-test --mode 'cluster' --run-dir 0_stateless --skip '13_0004_q4'

```

Reply:
The ErrorCode should be lazy initialized unless there are some errors captured in expression framework.